lib/resque_bus/server.rb

```diff
@@ -22,6 +22,123 @@ def self.included(base)
         
       }
     end
+
+    class Helpers
+      class << self
+        def parse_query(query_string)
+          has_open_brace = query_string.include?("{")
+          has_close_brace = query_string.include?("}")
+          has_multiple_lines = query_string.include?("\n")
+          has_colon = query_string.include?(":")
+          has_comma = query_string.include?(",")
+          has_quote = query_string.include?("\"")
+
+          exception = nil
+
+          # first let's see if it parses
+          begin
+            query_attributes = JSON.parse(query_string)
+            raise "Not a JSON Object" unless query_attributes.is_a?(Hash)
+          rescue Exception => e
+            exception = e
+          end
+          return query_attributes unless exception
+
+          if query_attributes
+            # it parsed but it's something else
+            if query_attributes.is_a?(Array) && query_attributes.length == 1
+              # maybe it's the thing from the queue
+              json_string = query_attributes.first
+              fixed = JSON.parse(json_string) rescue nil
+              return fixed if fixed
+            end
+
+            # something else?
+            raise exception
+          end
+
+          if !has_open_brace && !has_close_brace
+            # maybe they just forgot the braces
+            fixed = JSON.parse("{ #{query_string} }") rescue nil
```

Comment:
Couldn't we do this on line 38 before we parse? If we're concerned about the possibility that it's an array, maybe we can also look at the opening/closing brackets to see if they are square braces.

Reply:
One thing I tried to do was not mess with the input if it would parse. That's why I parsed first. Seems like a safe practice - if they put something that works use it before messing around with it.